front/src/applications/editor/components/LinearMetadata/FormComponent.tsx

```diff
@@ -427,4 +423,54 @@ export const FormComponent: React.FC<FieldProps> = (props) => {
   );
 };
 
+export const FormComponent: React.FC<FieldProps> = (props) => {
+  const { name, formContext, schema, registry } = props;
+  const Fields = utils.getDefaultRegistry().fields;
+
+  // Get the distance of the geometry
+  const distance = useMemo(() => {
+    if (!isNil(formContext.length)) {
+      return formContext.length;
+    }
+    if (formContext.geometry?.type === 'LineString') {
+      return getLineStringDistance(formContext.geometry);
+    }
+    return 0;
+  }, [formContext]);
+
+  // Remove the 'valueField' required field because it is required by the backend. However,
+  // the segment with missing values is filtered in 'customOnChange' before being sent to the backend,
+  // and then re-added by 'fixLinearMetadataItems'.
+  const requiredFilter = (requireds: string[]) =>
+    requireds.filter((r) => ['end', 'begin'].includes(r));
+
+  // Compute the JSON schema of the linear metadata item
+  const jsonSchema = useMemo(
+    () =>
+      getFieldJsonSchema(
+        schema,
+        registry.rootSchema,
+        requiredFilter,
+        distance
+          ? {
+              begin: {
+                minimum: 0,
+                maximum: distance,
+              },
+              end: {
+                minimum: 0,
+                maximum: distance,
+              },
+            }
+          : {}
+      ),
+    [schema, registry.rootSchema, distance]
+  );
+
+  if (!LINEAR_METADATA_FIELDS.includes(name))
+    return <Fields.ArrayField {...props} schema={jsonSchema} />;
+
+  return <IntervalEditorComponent {...props} />;
```

Comment:
Can you add some props to avoid duplication ? 
```suggestion
  return (
    <IntervalEditorComponent
      jsonSchema={jsonSchema}
      distance={distance}
      requiredFilter={requiredFilter}
      {...props}
    />
  );
```